### ISSUE TITLE
Inlay Hints style like JetBrains Soft

### ISSUE THREAD

**Issue by NEX-S** [lvimuser/lsp-inlayhints.nvim#17]
is there any options can make inlay hints like jetbrains intelij stuff? like in C programming 
printf("Hello") will become printf(`_Format:`"Hello")

**Comment by NEX-S** [comment]
Btw how to remove the bracket like `( )` around the inlay hints....

**Comment by lvimuser** [comment]
> is there any options can make inlay hints like jetbrains intelij stuff? like in C programming printf("Hello") will become printf(`_Format:`"Hello")

We're blocked by [anticonceal](https://github.com/neovim/neovim/pull/9496).

> Btw how to remove the bracket like `( )` around the inlay hints....

I'll think about it, do note that this makes easier to differentiate between hints. See also https://github.com/lvimuser/lsp-inlayhints.nvim/pull/13#issuecomment-1221409211.

**Comment by zsmatrix62** [comment]
I got the error today by using the branch mentioned:

<img width="1288" alt="image" src="https://user-images.githubusercontent.com/2951879/189507544-6083a0bc-a5b3-45d9-a3fe-8eb0fe9eb74a.png">

**Comment by lvimuser** [comment]
> I got the error today by using the branch mentioned:

The PR hasn't landed yet, you have to build neovim from source and manually apply the patch.

**Comment by zsmatrix62** [comment]
> 

actually I'm pulling the latest source code from vim repo and built on myself.

**Comment by lvimuser** [comment]
> > 
> 
> actually I'm pulling the latest source code from vim repo and built on myself.

Did you also apply the patch?
`wget https://patch-diff.githubusercontent.com/raw/neovim/neovim/pull/20130.patch && git apply 20130.patch`

**Comment by ecosse3** [comment]
@zsmatrix62 @lvimuser How did you enable that after applying the patch to neovim repository and building it on yourself? Do I need any special configuration after building? I can still see the old virtual text behavior.

**Comment by ratheesh** [comment]
With clangd, I observe that in the case of nested function call, I see that only inner call has inlay-hints and not the outer call.
![image](https://user-images.githubusercontent.com/70958/194712850-9433c03b-2691-4220-982a-e61ec96921f0.png)

Can this be `neovim/treesitter/LSP` issue? Can this be address in this plugin?

**Comment by lvimuser** [comment]
> With clangd, I observe that in the case of nested function call, I see that only inner call has inlay-hints and not the outer call. 
> Can this be `neovim/treesitter/LSP` issue? Can this be address in this plugin?

Try https://github.com/lvimuser/lsp-inlayhints.nvim/issues/11#issuecomment-1218269707 on the given range and check the output.

**Comment by Baehn** [comment]
applying the patch on current neovim results in errors, I tried to merge it but get errors on compiling

**Comment by willothy** [comment]
Awesome, works for me! Looks essentially the same as vscode's inlay hints now. I did have to fix one thing when patching though: the patch includes a call to nonexistent `STRLEN` on line 1649 of `src/nvim/drawline.c`. I changed that to `strlen`, and it compiled just fine.

**Comment by zsmatrix62** [comment]
> Awesome, works for me! Looks essentially the same as vscode's inlay hints now. I did have to fix one thing when patching though: the patch includes a call to nonexistent macro `STRLEN` at `src/nvim/drawline.c:1649`. I changed that to `strlen`, and it compiled just fine.

Hello @willothy , could you please give a sh script from compelling vim to applying the patch  ... I'm quite not familiar with the setup ... much appreciated ..

**Comment by willothy** [comment]
@zsmatrix62 Sure! Here's what I did. If you don't want to do it yourself, the "active" branch in my neovim fork is already patched if you just want to clone that.

    #!/bin/sh

    # Ensure you have basic C tools installed (make, cmake, gcc, etc.)
    # Ensure you have the github CLI installed

    # clone the neovim repo
    gh repo clone neovim/neovim       # or git clone git@github.com:neovim/neovim.git

    # cd into project root
    cd neovim

    # checkout the patch PR
    
    gh pr checkout 20130              # or git checkout a59e09c

    # create a new branch with the patch
    git checkout -b patch-20130

    # checkout the latest nightly neovim
    git checkout nightly

    # merge the changes from the patch branch
    git merge patch-20130

    # fix any merge conflicts 
    # there should only be some in src/nvim/drawline.c and some in test/functional/ui/decorations_spec.lua
    # accept incoming (patch-20130) changes in drawline.c
    # accept HEAD (nightly) changes in test/functional/ui/decorations_spec.lua

    # before you commit the merge
    # go to src/nvim/drawline.c in your editor, find line 1649.
    # It should look like this:
    #    n_extra = (int)STRLEN(p_extra);
    # change it to this:
    #    n_extra = (int)strlen(p_extra);

    # commit the changes
    git add src/nvim/drawline.c
    git add test/functional/ui/decorations_spec.lua
    git commit -m "Merge inline hints patch"

    # Build neovim
    make CMAKE_BUILD_TYPE=RelWithDebInfo
    sudo make install

**Comment by zsmatrix62** [comment]
> @zsmatrix62 Sure! Here's what I did. If you don't want to do it yourself, the "active" branch in my neovim fork is already patched if you just want to clone that.
> 
> 
> 
>     #!/bin/sh
> 
> 
> 
>     # Ensure you have basic C tools installed (make, cmake, gcc, etc.)
> 
>     # Ensure you have the github CLI installed
> 
> 
> 
>     # clone the neovim repo
> 
>     gh repo clone neovim/neovim       # or git clone git@github.com:neovim/neovim.git
> 
> 
> 
>     # cd into project root
> 
>     cd neovim
> 
> 
> 
>     # checkout the patch PR
> 
>     
> 
>     gh pr checkout 20130              # or git checkout a59e09c
> 
> 
> 
>     # create a new branch with the patch
> 
>     git checkout -b patch-20130
> 
> 
> 
>     # checkout the latest nightly neovim
> 
>     git checkout nightly
> 
> 
> 
>     # merge the changes from the patch branch
> 
>     git merge patch-20130
> 
> 
> 
>     # fix any merge conflicts 
> 
>     # there should only be some in src/nvim/drawline.c and some in test/functional/ui/decorations_spec.lua
> 
>     # accept incoming (patch-20130) changes in drawline.c
> 
>     # accept HEAD (nightly) changes in test/functional/ui/decorations_spec.lua
> 
> 
> 
>     # before you commit the merge
> 
>     # go to src/nvim/drawline.c in your editor, find line 1649.
> 
>     # It should look like this:
> 
>     #    n_extra = (int)STRLEN(p_extra);
> 
>     # change it to this:
> 
>     #    n_extra = (int)strlen(p_extra);
> 
> 
> 
>     # commit the changes
> 
>     git add src/nvim/drawline.c
> 
>     git add test/functional/ui/decorations_spec.lua
> 
>     git commit -m "Merge inline hints patch"
> 
> 
> 
>     # Build neovim
> 
>     make CMAKE_BUILD_TYPE=RelWithDebInfo
> 
>     sudo make install
> 
> 

Fantastic！thanks！

**Comment by lvimuser** [comment]
closing in favor of #46
